src/monodroid/jni/embedded-assemblies-zip.cc

```diff
@@ -129,25 +120,31 @@ EmbeddedAssemblies::zip_load_entries (int fd, const char *apk_name, monodroid_sh
 			exit (FATAL_EXIT_MISSING_ZIPALIGN);
 		}
 
+#if defined (DEBUG)
 		const char *last_slash = utils.find_last (entry_name, '/');
 		bool entry_is_overridden = last_slash == nullptr ? false : !should_register (last_slash + 1);
+#else
+		constexpr bool entry_is_overridden = false;
+#endif
 
-		if ((utils.ends_with (entry_name, ".pdb") || utils.ends_with (entry_name, ".mdb")) &&
-				register_debug_symbols &&
-				!entry_is_overridden &&
-				bundled_assembly_index >= 1) {
-			md_mmap_info map_info = md_mmap_apk_file(fd, data_offset, file_size, entry_name.get (), apk_name);
-			if (register_debug_symbols_for_assembly (entry_name, bundled_assemblies [bundled_assembly_index - 1], (const mono_byte*)map_info.area, static_cast<int>(file_size)))
-				continue;
+		if (register_debug_symbols && !entry_is_overridden && (utils.ends_with (entry_name, ".pdb") || utils.ends_with (entry_name, ".mdb"))) {
```

Comment:
In .NET 6, `.mdb` files won't even make it to the `.apk` anymore. Is it worth making this legacy-only?

Reply:
Yeah, I was wondering about that. If the `.mdb` files aren't a thing in NET6 I will definitely make that legacy only :)